include/dlaf/multiplication/general/impl.h

```diff
@@ -65,6 +65,82 @@ void General<B, D, T>::callNN(const blas::Op opA, const blas::Op opB, const T al
   }
 }
 
+template <Backend B, Device D, class T>
+void General<B, D, T>::callNN(common::Pipeline<comm::Communicator>& row_task_chain,
+                              common::Pipeline<comm::Communicator>& col_task_chain, const T alpha,
+                              MatrixRef<const T, D>& mat_a, MatrixRef<const T, D>& mat_b, const T beta,
+                              MatrixRef<T, D>& mat_c) {
+  namespace ex = pika::execution::experimental;
+
+  if (mat_c.size().isEmpty())
+    return;
```

Comment:
The case k=0 is not handled correctly.
The matrix C is not scaled accordingly to beta.